sympy/solvers/tests/test_solveset.py

```diff
@@ -798,6 +798,9 @@ def test_solve_trig():
     assert solveset(sin(y + a) - sin(y), a, domain=S.Reals) == \
         imageset(Lambda(n, 2*n*pi), S.Integers)
 
+    assert solveset_real(sin(2*x)*cos(x) + cos(2*x)*sin(x)-1, x) == \
+                            ConditionSet(x, Eq(sin(3*x) - 1, 0), S.Reals)
+
```

Comment:
Why returning a `ConditionSet` and not `ImageSet(Lambda(_n, 2*_n*pi/3 + pi/6), S.Integers)`? It seems that it could be solved by the code.

Reply:
`ConditionSet` is what `solve_trig1` returns. Although second helper function can solve this, it won't be called until former one returns an exception.

Reply:
Wish there was something which could decide which helper function would return a  better solution but I think it would slow down the whole process.

Reply:
> ConditionSet is what solve_trig1 returns

I don't think `solve_trig1` returns ConditionSet, it rather raises the error and it then goes to `_solve_trig2` where it gets solved (when I try locally it gets solved)

Reply:
Would this [untested suggestion] work at line 504 to return a solution for the test case `(sin(2*x)*cos(x) + cos(2*x)*sin(x) - 1, x)`?
```python
    sol1 = None
    try:
        sol1 = _solve_trig1(f, symbol, domain)
    except BaseException as error:
        pass
    if sol1 is None or isinstance(sol1, ConditionSet):
        try:
            sol = _solve_trig2(f, symbol, domain)
        except BaseException as error:
            pass
        if sol1 is not None and isinstance(sol, ConditionSet):
            if sol1.count_ops() < sol.count_ops():
                sol = sol1
    else:
        sol = sol1
    if sol is None:
        raise NotImplementedError(filldedent('''
            Solution to this kind of trigonometric equations
            is yet to be implemented'''))
    return sol
```

Reply:
That gives
```
>>> solveset(sin(2*x)*cos(x) + cos(2*x)*sin(x) - 1,x,S.Reals)
ImageSet(Lambda(_n, 2*_n*pi/3 + pi/6), S.Integers)
```
and it has only a single trig function (so it requires the modification for 1 trig arg). And the simplified form can't (presently) be solved by trig1:
```
>>> _solve_trig1(q,x,S.Reals)
Traceback (most recent call last):
...
TypeError: variable is not a symbol: -_y**6*I - 2*_y**3 + I
>>> _solve_trig2(q,x,S.Reals)
ImageSet(Lambda(_n, 2*_n*pi/3 + pi/6), S.Integers)
```

Reply:
In my branch 
```
>>> _solve_trig1(sin(2*x)*cos(x) + cos(2*x)*sin(x) - 1,x,S.Reals)
ConditionSet(x, Eq(sin(3*x) - 1, 0), S.Reals)
```
This [commit](https://github.com/sympy/sympy/pull/14455/files) solved it.

Reply:
With the help of above code, `solve_trig` is now returning a better answer instead of `ConditionSet`. I think it should be used.